editor/filesystem_dock.cpp

```diff
@@ -182,16 +182,38 @@ bool FileSystemDock::_create_tree(TreeItem *p_parent, EditorFileSystemDirectory
 	// Create a tree item for the subdirectory.
 	TreeItem *subdirectory_item = tree->create_item(p_parent);
 	String dname = p_dir->get_name();
+	String lpath = p_dir->get_path();
+
 	if (dname.is_empty()) {
 		dname = "res://";
 	}
 
+	// Set custom folder color (if applicable)
+	Dictionary custom_folder_colors = ProjectSettings::get_singleton()->get_setting("file_customization/folder_colors");
+	bool has_custom_color = custom_folder_colors.has(lpath);
+	Color custom_color = has_custom_color ? folder_colors[custom_folder_colors[lpath]] : Color();
+
 	subdirectory_item->set_text(0, dname);
 	subdirectory_item->set_structured_text_bidi_override(0, TextServer::STRUCTURED_TEXT_FILE);
 	subdirectory_item->set_icon(0, get_theme_icon(SNAME("Folder"), SNAME("EditorIcons")));
-	subdirectory_item->set_icon_modulate(0, get_theme_color(SNAME("folder_icon_color"), SNAME("FileDialog")));
+	if (has_custom_color) {
+		subdirectory_item->set_icon_modulate(0, custom_color);
+		subdirectory_item->set_custom_bg_color(0, custom_color.darkened(0.6));
+	} else {
+		TreeItem *parent = subdirectory_item->get_parent();
+		if (parent) {
+			Color parent_bg_color = parent->get_custom_bg_color(0);
+			if (parent_bg_color != custom_color) { // a.k.a parent_bg_color != Color()
```

Comment:
```suggestion
			if (parent_bg_color != Color()) {
```
I'd simplify this condition, as it's always equivalent, as per your comment